src/SessionMonitor/index.js

```diff
@@ -49,18 +51,38 @@ export class SessionMonitor {
 
   logoutUser() {
     // don't hit the logout endpoint over and over if the popup is already shown
-    if (this.popupShown) {
+    if (this.logoutPopupShown) {
       return;
     }
 
     getReduxStore().then((store) => {
       store.dispatch(logoutAPI(true));
-      this.popupShown = true;
+      this.logoutPopupShown = true;
+    });
+  }
+
+  warnUser(warnTime) {
+    // don't hit the logout endpoint over and over if the popup is already shown
+    if (this.inactivityWarningPopupShown) {
+      return;
+    }
+
+    getReduxStore().then((store) => {
+      store.dispatch({
+        type: 'UPDATE_POPUP',
+        data: {
+          inactivityWarningPopup: true,
+          inactivityWarningTime: warnTime,
+        },
+      });
+      this.inactivityWarningPopupShown = true;
```

Comment:
could both warning popup and logout popup be visible at the smae time and interference with each other?
if so maybe dismiss warning popup when the logout popup is activated?

Reply:
In the src/Popup/ReduxAuthTimeoutPopup.jsx code i have an if else statement insuring that the logged out popup always has priority over the warning and that they are never shown at the same time

Reply:
well i did but i think the code linter removed it (will look into why)

Reply:
it removed it because of the return no need for an else statement because second if will never run if first is true